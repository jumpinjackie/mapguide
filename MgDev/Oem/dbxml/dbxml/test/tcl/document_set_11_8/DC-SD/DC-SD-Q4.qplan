<XQuery>
  <FLWOR>
    <LetBinding name="item">
      <QueryPlanToAST>
        <NodePredicateFilterQP uri="" name="#tmp2">
          <StepQP axis="child" name="item" nodeType="element">
            <StepQP axis="child" name="catalog" nodeType="element">
              <ValueQP index="node-attribute-equality-string" operation="eq" child="id" value="I2"/>
            </StepQP>
          </StepQP>
          <ValueFilterQP comparison="eq" general="true">
            <StepQP axis="attribute" name="id" nodeType="attribute">
              <VariableQP name="#tmp2"/>
            </StepQP>
            <Sequence>
              <AnyAtomicTypeConstructor value="I2" typeuri="http://www.w3.org/2001/XMLSchema" typename="string"/>
            </Sequence>
          </ValueFilterQP>
        </NodePredicateFilterQP>
      </QueryPlanToAST>
    </LetBinding>
    <ForBinding name="prevItem">
      <Map>
        <QueryPlanToAST>
          <StepQP axis="child" name="catalog" nodeType="element">
            <SequentialScanQP nodeType="document"/>
          </StepQP>
        </QueryPlanToAST>
        <QueryPlanToAST>
          <NumericPredicateFilterQP>
            <PredicateFilterQP>
              <StepQP axis="child" name="item" nodeType="element">
                <ContextNodeQP/>
              </StepQP>
              <Operator name="node-order">
                <TreatAs>
                  <QueryPlanToAST>
                    <ContextNodeQP/>
                  </QueryPlanToAST>
                  <SequenceType occurrence="zero_or_one" testType="node"/>
                </TreatAs>
                <TreatAs>
                  <QueryPlanToAST>
                    <VariableQP name=":item"/>
                  </QueryPlanToAST>
                  <SequenceType occurrence="zero_or_one" testType="node"/>
                </TreatAs>
              </Operator>
            </PredicateFilterQP>
            <Operator name="equal">
              <Function name="{http://www.w3.org/2005/xpath-functions}:position"/>
              <Function name="{http://www.w3.org/2005/xpath-functions}:last"/>
            </Operator>
          </NumericPredicateFilterQP>
        </QueryPlanToAST>
      </Map>
    </ForBinding>
    <DOMConstructor type="element">
      <Name>
        <Sequence>
          <ATQNameConstructor uri="" prefix="" localname="Output" typeuri="http://www.w3.org/2001/XMLSchema" typename="QName"/>
        </Sequence>
      </Name>
      <Children>
        <DOMConstructor type="element">
          <Name>
            <Sequence>
              <ATQNameConstructor uri="" prefix="" localname="CurrentItem" typeuri="http://www.w3.org/2001/XMLSchema" typename="QName"/>
            </Sequence>
          </Name>
          <Children>
            <ContentSequence>
              <QueryPlanToAST>
                <StepQP axis="attribute" name="id" nodeType="attribute">
                  <VariableQP name=":item"/>
                </StepQP>
              </QueryPlanToAST>
            </ContentSequence>
          </Children>
        </DOMConstructor>
        <DOMConstructor type="element">
          <Name>
            <Sequence>
              <ATQNameConstructor uri="" prefix="" localname="PreviousItem" typeuri="http://www.w3.org/2001/XMLSchema" typename="QName"/>
            </Sequence>
          </Name>
          <Children>
            <ContentSequence>
              <QueryPlanToAST>
                <StepQP axis="attribute" name="id" nodeType="attribute">
                  <VariableQP name=":prevItem"/>
                </StepQP>
              </QueryPlanToAST>
            </ContentSequence>
          </Children>
        </DOMConstructor>
      </Children>
    </DOMConstructor>
  </FLWOR>
</XQuery>
