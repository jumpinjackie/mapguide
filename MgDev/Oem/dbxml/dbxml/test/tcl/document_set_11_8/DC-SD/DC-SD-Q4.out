<Output><CurrentItem id="I2"/><PreviousItem id="I1"/></Output>